### ISSUE TITLE
No private IPs for VMs created on OpenStack

### ISSUE THREAD

**Issue by lgiommi** [grycap/im#1641]
Hi,
In INFN Cloud, we are using the IM as the service that allows us to deploy VMs on OpenStack instances. However, we have a provider that does not use Neutron to manage floating IP pools (it should be fortigate). As noted in the code [here](https://github.com/grycap/im/blob/bcfc3466d6ecf20bbadbdfcef53bf4e33dee3dd2/IM/connectors/OpenStack.py#L1356), for the VMs we attempt to create with this provider, the private IP is not assigned, whereas the public IP is successfully assigned.

Is there a possible workaround to address this issue? For instance, could a configuration parameter be introduced to handle this scenario?

Thank you

**Comment by micafer** [comment]
Hi @lgiommi,

So in this case both the private and the public IPs are managed using floating IPs, right?

**Comment by lgiommi** [comment]
Hi @micafer,
yes also public IPs are managed using floating IPs

**Comment by micafer** [comment]
We can try to fix this issue.
It will be possible to have some account, may be some temporal application credentials that we may use to make some tests?

**Comment by lgiommi** [comment]
Thanks @micafer. I contacted a guy from that site and he should take charge of you request

**Comment by matpauranoo-infn** [comment]
Hi @micafer,
you can have an account on my institution's OpenStack by filling out the request available on [the dedicated page](https://www.recas-bari.it/index.php/en/recas-bari-servizi-en/richiesta-credenziali-2).
I will follow the authorization process internally.
Thanks.

**Comment by micafer** [comment]
HI @matpauranoo-infn,
Requested!

**Comment by matpauranoo-infn** [comment]
OK! Thanks!

**Comment by matpauranoo-infn** [comment]
Dear @micafer,
the last necessary step is to register on [our Identity and Access Management system](https://iam.recas.ba.infn.it).
Thank you.

**Comment by micafer** [comment]
Done!

**Comment by micafer** [comment]
Done, it was in the spam folder.

**Comment by matpauranoo-infn** [comment]
@micafer  your access to services is working ...

**Comment by micafer** [comment]
It could be feasible that you add a tag named "enable_two_nics" to one of the nets of your site?

**Comment by matpauranoo-infn** [comment]
Tag added to "private_subnet".

**Comment by micafer** [comment]
could you try using "ghcr.io/grycap/im:1.17.1-4" image?
With this tag added in the network it must work.

**Comment by giosava94** [comment]
Hi, we tried using that image, but it does not use the latest version of the IM. It downloads version 1.17.1 of the IM pip package released on October the 31st. That package does not contain your changes about this topic. Can you fix this behavior?

**Comment by giosava94** [comment]
Tahnk you for linking the commit with your changes. 

We saw the commit you referenced, and we are using the docker image `ghcr.io/grycap/im:1.17.1-4`. 

Though, when entering the container, and running the command `/usr/local/bin/im_service --version`, the ouput is **1.17.1**.and, in fact, looking at the dockerfile the pypi version of the IM pip package installed is 1.17.1 (which does not contain the target commit). Looking on the [IM Pip package documentation](https://pypi.org/project/IM/#history) that version was released on October.

Moreover, looking at the logs, we saw from the logs that the message is still the one before your fix.

I hope this helps for better explaining the problem and resolving the issue.
